sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md

```diff
@@ -0,0 +1,502 @@
+
+# Mocking Client Types
+
+Event Hubs is built to support unit testing with mocks, as described in the [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking). This is an important feature of the library that allows developers to write tests that are completely focused on their own application logic, though they depend on the Event Hubs types.
+
+The examples that follow focus on scenarios likely to occur in applications, and demonstrate how to mock the Event Hubs types typically used in each scenario.
+
+## Publishing Events with the `EventHubProducerClient`
```

Comment:
```suggestion
## Publishing events with the `EventHubProducerClient`
```

nit: We've been asked to keep to sentence-style capitalization in our Markdown headings to conform to the [Microsoft Style Guide](https://docs.microsoft.com/en-us/style-guide/capitalization#sentence-style-capitalization-in-titles-and-headings).